packages/bpk-component-slider/src/BpkSlider.tsx

```diff
@@ -117,20 +126,67 @@ const BpkSlider = ({
 
 // Work around until radix-ui/react-slider is updated to accept an inputRef prop https://github.com/radix-ui/primitives/pull/3033
 const BubbleInput = forwardRef(
-  (props: ComponentPropsWithRef<'input'>, forwardedRef) => {
-    const { value, ...inputProps } = props;
+  (
+    props: ComponentPropsWithRef<'input'> & {
+      thumbRef: RefObject<HTMLElement>;
+    },
+    forwardedRef,
+  ) => {
+    const { thumbRef, value, ...inputProps } = props;
     const ref = useRef<HTMLInputElement>();
     const composedRefs = useComposedRefs(forwardedRef, ref);
 
-    const prevValue = usePrevious(value);
-
-    // Bubble value change to parents (e.g form change event)
+    // Provide native behaviour that the input range type provides.
     useEffect(() => {
-      const input = ref.current!;
-      if (prevValue !== value) {
-        setNativeValue(input, `${value}`);
+      const thumb = thumbRef.current;
+      const input = ref.current;
+      if (thumb) {
+        const interactionEndHandler = (prevValue: number) => {
+          if (input) {
+            const newValue = parseFloat(input.getAttribute('value') as string);
+            if (prevValue !== newValue) {
+              setNativeValue(input, newValue);
+            }
```

Comment:
Why do we parse things as a float? 
Why do we need to check the previous value vs the new?

Reply:
> Why do we parse things as a float?
The input `type` is a 'number' and though we could use `parseInt` the range can actually support a float value. 
When we parse a string "3" as float we still get a whole number 3 rather than 3.0. And as were not parsing hex strings. Float fits this purpose really nicely. Defaulting to an int when it's a whole number.

> Why do we need to check the previous value vs the new?
If a user interacts and doesn't move the value then there is no "change" so we don't fire an event.

Reply:
What if the slider is `strong`, `medium`, `light` or do sliders not work like that? https://www.skyscanner.design/latest/components/slider/web-aNXvlY7y#section-non-integer-values-54 I think it doesn't work like that. 

Always makes me a little nervous when JS used in this way to have happy coincidences. [Math.trunc](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Math/trunc) might be an alternative. 

Why should we not care about the 10ths/100th etc within the float like value returned? Would it cause problems if we did `3.1 === 3.1`?